Tests/web3swiftTests/localTests/BIP44Tests.swift

```diff
@@ -0,0 +1,188 @@
+            XCTAssertTrue(true)
+            XCTAssertEqual(mockTransactionChecker.addresses, accountZeroScannedAddresses)
+        }
+    }
+
+    func testAccountOneWithTransactionsInAccountZeroNotWarns() async throws {
+        do {
+            let rootNode = try rootNode()
+            let path = "m/44'/60'/1'/0/0"
+            var results = false.times(n: 19)
+            results.append(true)
+            mockTransactionChecker.results = results
+
+            let childNode = try await rootNode.derive(path: path, warns: true, transactionChecker: mockTransactionChecker)
+
+            XCTAssertEqual(try XCTUnwrap(childNode).publicKey.toHexString(), "036cd8f1bad46fa7caf7a80d48528b90db2a3b7a5c9a18d74d61b286e03850abf4")
+            XCTAssertEqual(mockTransactionChecker.addresses, accountZeroScannedAddresses)
+        } catch BIP44Error.warning {
+            XCTFail("BIP44Error.warning must not be thrown")
+        }
+    }
+
+    func testAccountTwoWithTransactionsInAccountZeroButNotInOneWarns() async throws {
+        do {
+            let rootNode = try rootNode()
+            let path = "m/44'/60'/2'/0/0"
+            var results: [Bool] = .init()
+            results.append(true)
+            results.append(contentsOf: false.times(n: 20))
+            mockTransactionChecker.results = results
+
+            _ = try await rootNode.derive(path: path, warns: true, transactionChecker: mockTransactionChecker)
+
+            XCTFail("Child must not be created usign warns true for the path: \(path)")
+        } catch BIP44Error.warning {
+            XCTAssertTrue(true)
+            XCTAssertEqual(mockTransactionChecker.addresses, accountZeroAndOneScannedAddresses)
+            XCTAssertEqual(mockTransactionChecker.addresses.count, 21)
+        }
+    }
+
+    // MARK: - change + addressIndex
+
+    func testAccountOneAndInternalAndNotZeroAddressIndexWithTransactionsInAccountZeroNotWarns() async throws {
+        do {
+            let rootNode = try rootNode()
+            let path = "m/44'/60'/1'/1/128"
+            var results = false.times(n: 19)
+            results.append(true)
+            mockTransactionChecker.results = results
+
+            let childNode = try await rootNode.derive(path: path, warns: true, transactionChecker: mockTransactionChecker)
+
+            XCTAssertEqual(try XCTUnwrap(childNode).publicKey.toHexString(), "0282134e44d4c040a4b4c1a780d8302955096cf1d5e738b161c83f0ce1b863c14e")
+            XCTAssertEqual(mockTransactionChecker.addresses, accountZeroScannedAddresses)
+        } catch BIP44Error.warning {
+            XCTFail("BIP44Error.warning must not be thrown")
+        }
+    }
+
+    // MARK: - private
+
+    private func rootNode() throws -> HDNode {
+        let mnemonic = "fruit wave dwarf banana earth journey tattoo true farm silk olive fence"
+        let seed = try XCTUnwrap(BIP39.seedFromMmemonics(mnemonic, password: ""))
+        return try XCTUnwrap(HDNode(seed: seed))
+    }
+}
+
+extension Bool {
```

Comment:
Please, mark it as `fileprivate`.

Reply:
do you think that I should move it to a diff file?

Reply:
Not necessarily but feel free to do so. A file named something like `Extensions.swift` will be good enough as it's within `Tests` package. Should not generate any misunderstanding.

Reply:
changed to private but not moved to its own file